test/Microsoft.Health.Dicom.Tests.Integration/Persistence/SqlServerSchemaUpgradeTests.cs

```diff
@@ -34,5 +36,13 @@ public async Task GivenTwoSchemaInitializationMethods_WhenCreatingTwoDatabases_B
             await snapshotFixture.DisposeAsync();
             await diffFixture.DisposeAsync();
         }
+
+        [Fact]
+        public async Task GivenASchemaVersion_WhenApplyingDiffTwice_ShouldSucceed()
+        {
+            SqlDataStoreTestsFixture snapshotFixture = new SqlDataStoreTestsFixture(SqlDataStoreTestsFixture.GenerateDatabaseName("SNAPSHOT"));
+            await snapshotFixture.InitializeAsync(forceIncrementalSchemaUpgrade: false);
+            await snapshotFixture.SchemaUpgradeRunner.ApplySchemaAsync(SchemaVersionConstants.Max, applyFullSchemaSnapshot: false, CancellationToken.None);
```

Comment:
I think the test in FHIR is similar to this, but I think we can improve it by making this test a Theory that will the test multiple times, for each schema version between SchemaVersionConstants.Min and Max.